pkg/ccl/backupccl/manifest_handling.go

```diff
@@ -1259,18 +1245,54 @@ func readLatestCheckpointFile(
 	// First try reading from the progress directory. If the backup is from
 	// an older version, it may not exist there yet so try reading
 	// in the base directory if the first attempt fails.
-	r, err := exportStore.ReadFile(ctx, backupProgressDirectory+"/"+filename)
-	if err != nil {
+	var checkpoint string
+	var checkpointFound bool
+	var r ioctx.ReadCloserCtx
+	var err error
+
+	// HTTP doesn't support listing so only try reading the non-timestamped
+	// version of the filename in the progress directory.
+	if exportStore.Conf().Provider == roachpb.ExternalStorageProvider_http {
+		r, err = exportStore.ReadFile(ctx, backupProgressDirectory+"/"+filename)
+		if err == nil {
```

Comment:
hmm is this a bug? If err != nil we want to return the error here.

Reply:
So basically if it's http, we can't list, so it just tries to read non ts version from progress directory. Otherwise if it isn't http, then we try to read the ts version using list.

If the http fails, it could be because we didn't find the file/directory, so we need to retry in the base directory. However, if there wasn't an error, then we don't need to retry, just return the reader.

If the non http list fails, it's because of a non file/directory not found error. Since list won't fail if it doesn't find anything. So thats why we don't have to retry the base directory in this case and can just return the error.

I added a comment to try and make it more clear, and am returning nil now to make it more obvious I meant to return that.